satpy/readers/sar_c_safe.py

```diff
@@ -704,3 +689,85 @@ def start_time(self):
+        """Load some data."""
+        if kwargs:
+            raise NotImplementedError(f"Don't know how to handle kwargs {kwargs}")
+        datasets = DatasetDict()
+        for key in dataset_keys:
+            for handler in self.storage_items.values():
+                val = handler.get_dataset(key, info=dict())
+                if val is not None:
+                    val.attrs["start_time"] = handler.start_time
+                    # val.attrs["footprint"] = self.footprint
+                    if key["name"] not in ["longitude", "latitude"]:
+                        lonlats = self.load([DataID(self._id_keys, name="longitude", polarization=key["polarization"]),
+                                             DataID(self._id_keys, name="latitude", polarization=key["polarization"])])
+                        gcps = val.coords["spatial_ref"].attrs["gcps"]
+                        from pyresample.future.geometry import SwathDefinition
+                        val.attrs["area"] = SwathDefinition(lonlats["longitude"], lonlats["latitude"],
+                                                            attrs=dict(gcps=gcps))
+                    datasets[key] = val
+                    continue
+        return datasets
+
+    def create_storage_items(self, files, **kwargs):
+        """Create the storage items."""
+        filenames = files
+        files_by_type = defaultdict(list)
+        for file_type, type_info in self.config["file_types"].items():
+            files_by_type[file_type].extend(self.filename_items_for_filetype(filenames, type_info))
+        image_shapes = dict()
+        for annotation_file, annotation_info in files_by_type["safe_annotation"]:
+            annotation_fh = SAFEXMLAnnotation(annotation_file,
+                                              filename_info=annotation_info,
+                                              filetype_info=None)
+            image_shapes[annotation_info["polarization"]] = annotation_fh.image_shape
+
+        calibration_handlers = dict()
+        for calibration_file, calibration_info in files_by_type["safe_calibration"]:
+            polarization = calibration_info["polarization"]
+            calibration_handlers[polarization] = Calibrator(calibration_file,
+                                                            filename_info=calibration_info,
+                                                            filetype_info=None,
+                                                            image_shape=image_shapes[polarization])
+
+        noise_handlers = dict()
+        for noise_file, noise_info in files_by_type["safe_noise"]:
+            polarization = noise_info["polarization"]
+            noise_handlers[polarization] = Denoiser(noise_file,
+                                                    filename_info=noise_info,
+                                                    filetype_info=None,
+                                                    image_shape=image_shapes[polarization])
```

Comment:
This could be a separate private method returning `noise_handlers`.

Reply:
Fixed in [`f0c382e` (#2727)](https://github.com/pytroll/satpy/pull/2727/commits/f0c382e8822e7dcbacdd1a85cb67faceeaf7b0e4)